### PR TITLE
[5.1] Better way to change faker locale

### PR DESCRIPTION
**Pull request by vluzrmos**

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/FactoryBuilder.php [hunk 1]
```diff
@@ -45,16 +45,16 @@ class FactoryBuilder
...
      * Create an new builder instance.
      *
-     * @param  string  $class
-     * @param  string  $name
-     * @param  array  $definitions
-     * @return void
+     * @param  string $class
+     * @param  string $name
+     * @param  array $definitions
+     * @param \Faker\Generator $faker
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/FactoryBuilder.php]: cs

File: src/Illuminate/Database/Eloquent/Factory.php [hunk 2]
```diff
@@ -196,4 +211,46 @@ public function offsetUnset($offset)
     {
         unset($this->definitions[$offset]);
     }
+
+    /**
+     * Get a new instance of faker locale or the latest created
+     *
+     * @param string|null $locale
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Factory.php]: cs

File: src/Illuminate/Database/Eloquent/Factory.php [hunk 3]
```diff
@@ -16,17 +16,32 @@ class Factory implements ArrayAccess
...
 
     /**
+     * Default locale used on faker
+     *
+     * @var string|null
+     */
+    protected $fakerLocale = null;
+
+    /**
+     * @var \Faker\Generator|null
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Factory.php]: needs a short description

File: src/Illuminate/Database/Eloquent/Factory.php [hunk 4]
```diff
@@ -16,17 +16,32 @@ class Factory implements ArrayAccess
     protected $definitions = [];
 
     /**
+     * Default locale used on faker
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Factory.php]: cs

File: src/Illuminate/Database/Eloquent/Factory.php [hunk 5]
```diff
@@ -16,17 +16,32 @@ class Factory implements ArrayAccess
...
+     *
+     * @var string|null
+     */
+    protected $fakerLocale = null;
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Factory.php]: don't need `= null;` here

File: src/Illuminate/Database/Eloquent/Factory.php [hunk 6]
```diff
@@ -16,17 +16,32 @@ class Factory implements ArrayAccess
...
+
+    /**
+     * @var \Faker\Generator|null
+     */
+    protected $faker = null;
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Factory.php]: same

File: src/Illuminate/Database/Eloquent/FactoryBuilder.php [hunk 7]
```diff
@@ -45,16 +45,16 @@ class FactoryBuilder
...
      */
-    public function __construct($class, $name, array $definitions)
+    public function __construct($class, $name, array $definitions, \Faker\Generator $faker = null)
```
Comment (gmsantosxl) [inline src/Illuminate/Database/Eloquent/FactoryBuilder.php]: The second parameter is expecting a Faker\Generator instance, but a string is given at DatabaseServiceProvider:59

**Comment by vluzrmos** [comment]
@GramhamCampbell I made some fixes.

File: src/Illuminate/Database/Eloquent/Factory.php (hunk 4 above)
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Factory.php]: still need fixing. Missing fullstop...

**Comment by vluzrmos** [comment]
@GrahamCampbell 
What do you think about make the helper `factory` returns the singleton if called without parameters?

That will allow to do:
```php
factory()->setLocale('en_US');
```
on any line of a model factory and all lines after it will have that locale.

**Comment by taylorotwell** [comment]
You can just create your own faker instance in your model factories.

**Pull request closed without merging**